app/main.py

```diff
@@ -1,8 +1,22 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    for result in people:
+        Person(result["name"], result["age"])
+    for result in people:
+        result_people = Person.people[result["name"]]
+        if "wife" in result and result["wife"] is not None:
+            result_people.wife = Person.people[result["wife"]]
```

Comment:
The code assumes that the `wife` key will always correspond to a valid name in the `Person.people` dictionary. This can lead to a `KeyError` if the name is not found. Consider adding a check to ensure the name exists in the dictionary before accessing it.